entrypoint/src/mill/entrypoint/MillBootstrapModule.scala

```diff
@@ -0,0 +1,214 @@
+package mill.entrypoint
+import coursier.{Dependency, Module, Organization}
+
+import collection.mutable
+import mill._
+import mill.api.{Loose, PathRef, Result}
+import mill.define.Task
+import mill.scalalib.{BoundDep, DepSyntax, Lib, Versions}
+class MillBootstrapModule(enclosingClasspath: Seq[os.Path], base: os.Path)
+  extends mill.define.BaseModule(base)(implicitly, implicitly, implicitly, implicitly, mill.define.Caller(())) {
+
+  implicit lazy val millDiscover: _root_.mill.define.Discover[this.type] = _root_.mill.define.Discover[this.type]
+
+  object millbuild extends mill.scalalib.ScalaModule {
```

Comment:
The choice of putting the logic inside a nested `millbuild` module, instead of the enclosing `MillBootstrapModule`, is somewhat arbitrary. Not sure which way is better

Reply:
In the future, it might be interesting to experiment with a different file structure that is faster to compile.
But for this PRs purpose, anything that works as before looks good to me.